speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/DeveloperMenu.cpp

```diff
@@ -0,0 +1,174 @@
+    char dst[512];
+    sprintf(dst, "%s%s", GfLocalDir(), dstStr.c_str());
+    void* readParam = GfParmReadFile(dst, GFPARM_RMODE_REREAD | GFPARM_RMODE_CREAT);
+
+    // Write sync type
+    char val[32];
+    sprintf(val, "%d", m_sync);
+    GfParmSetStr(readParam, PRM_SYNC, GFMNU_ATTR_SELECTED, val);
+
+    // Write recorder status
+    GfParmSetStr(readParam, PRM_RECORD_TGGLE, GFMNU_ATTR_CHECKED, GfuiMenuBoolToStr(m_replayRecorderOn));
+
+    // Write queued changes
+    GfParmWriteFile(nullptr, readParam, "DeveloperMenu");
+}
+
+/// @brief Saves the settings so the mediator (or future instances) can access them
+static void SaveSettings()
+{
+    SMediator* mediator = SMediator::GetInstance();
+
+    mediator->SetSyncOption(m_sync == 1);
+    mediator->SetReplayRecorderSetting(m_replayRecorderOn);
+
+    SaveSettingsToFile();
+}
+
+static void OnActivate(void* /* dummy */)
+{
+    LoadSettings();
+    SynchronizeControls();
+}
+
+/// @brief Switches back to the reseacher menu
+static void SwitchToResearcherMenu(void* /* dummy */)
+{
+    // go back to the researcher screen
+    GfuiScreenActivate(s_prevHandle);
+}
+
+/// @brief Saves the settings and then switched back to the researcher menu.
+static void SaveAndGoBack(void* /* dummy */)
+{
+    SaveSettings();
+
+    SwitchToResearcherMenu(nullptr);
+}
+
+/// @brief        Sets the type of synchronization
+/// @param p_info Information about the radio button list
+static void SelectSync(tRadioButtonInfo* p_info)
+{
+    m_sync = (SyncType)p_info->Selected;
+}
+
+/// @brief        Enables/disables the replay recorder
+/// @param p_info Information on the checkbox
+static void SelectRecorderOnOff(tCheckBoxInfo* p_info)
+{
+    m_replayRecorderOn = p_info->bChecked;
+}
+
+void* DeveloperMenuInit(void* p_prevMenu)
+{
+    // screen already created
+    if (s_scrHandle) return s_scrHandle;
+
+    s_scrHandle = GfuiScreenCreate((float*)nullptr, nullptr, OnActivate, nullptr,
+                                   (tfuiCallback) nullptr, 1);
+    s_prevHandle = p_prevMenu;
+
+    void* param = GfuiMenuLoad("DeveloperMenu.xml");
+    GfuiMenuCreateStaticControls(s_scrHandle, param);
+    // add UI button controls here
+    GfuiMenuCreateButtonControl(s_scrHandle, param, "CancelButton",
+                                s_scrHandle, SwitchToResearcherMenu);
+    GfuiMenuCreateButtonControl(s_scrHandle, param, "ApplyButton",
+                                s_scrHandle, SaveAndGoBack);
+    m_syncButtonList = GfuiMenuCreateRadioButtonListControl(s_scrHandle, param, PRM_SYNC, nullptr, SelectSync);
+    m_replayRecorder = GfuiMenuCreateCheckboxControl(s_scrHandle, param, PRM_RECORD_TGGLE, nullptr, SelectRecorderOnOff);
+
+    GfParmReleaseHandle(param);
+    // add keyboard controls here
+    GfuiAddKey(s_scrHandle, GFUIK_RETURN, "Apply", nullptr, SaveAndGoBack, nullptr);
+    GfuiAddKey(s_scrHandle, GFUIK_ESCAPE, "Cancel", s_prevHandle,
+               SwitchToResearcherMenu, nullptr);
+
```

Comment:
add `GfuiMenuDefaultKeysAdd(s_scrHandle);` to make the menu navigatable via arrow keys.